### PR TITLE
Preliminary work for introducing the new module function design pattern

### PR DESCRIPTION
**Pull request by oOKexOo**
**New Design Patterns:**
- See [conventions.txt](https://github.com/ArmaAchilles/Achilles/blob/AddonRestructuring/conventions.txt)

**When merged this pull request will:**
- [x] Update attributes of `cfgPatches`.
- [x] Add macros for functions and modules.
- [x] Add `modules_f` and `functions_f` folders, which will replace the other module and function folders in future. Revised modules with the new module function design pattern will be moved to `modules_f`.
- [ ] Provide functions for a new module function design pattern that entirely resides in unscheduled environment and that makes the use of curator event handlers more straightforward.
    - `Achilles_fnc_module_init` (handles the different cases of module placements).
    - `Achilles_fnc_module_openDialog` (unscheduled analogue of `Achilles_fnc_showChooseDialog`).
    - `Achilles_fnc_module_selectionOption` (unscheduled analogue of `Achilles_fnc_SelectUnits`).
- [ ] Adds event handler needed for #403 and add lines to curator event handlers that call the module function when `3den=1`.

### CODE REVIEW

**Comment by CreepPork** [comment]
I think that the conventions should be in markdown and possibly added to the GitHub Wiki later.

File: @AresModAchillesExpansion/addons/ui_f/config.cpp [hunk 1]
```diff
@@ -20,12 +22,13 @@ class CfgPatches
...
-			"achilles_data_f_achilles",
-			"achilles_data_f_ares",
-            "achilles_settings_f"
+			"Achilles_Language_F",
+			"Achilles_Functions_F",
+			"Achilles_Functions_F_Ares",
+			"Achilles_Functions_F_Achilles",
+			"Achilles_Data_F_Achilles",
+			"Achilles_Data_F_Ares",
+            "Achilles_Settings_F"
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/ui_f/config.cpp]: Weird spacing.

File: @AresModAchillesExpansion/addons/modules_f_achilles/config.cpp [hunk 2]
```diff
@@ -122,11 +122,11 @@ class CfgPatches
...
-			"achilles_language_f",
-			"achilles_functions_f_ares",
-			"achilles_functions_f_achilles",
-			"achilles_data_f_achilles",
-			"achilles_data_f_ares"
+			"achilles_Language_F",
+			"achilles_Functions_F_Ares",
+			"achilles_Functions_F_Achilles",
+			"achilles_Data_F_Achilles",
+			"achilles_Data_F_Ares"
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f_achilles/config.cpp]: Inconsistent naming.

File: @AresModAchillesExpansion/addons/modules_f/cfgVehicles.hpp [hunk 3]
```diff
@@ -0,0 +1,37 @@
+class CfgVehicles
+{
+	class Logic;
+	class Module_F: Logic
+	{
+		class ModuleDescription;
+	};
+	class MODULE_BASE_ACHIL : Module_F
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/modules_f/cfgVehicles.hpp]: Why is this class name capitalized?

Comment (CreepPork) [reply @AresModAchillesExpansion/addons/modules_f/cfgVehicles.hpp]: This doesn't follow Arma 3 conventions.

File: @AresModAchillesExpansion/addons/modules_f_ares/config.cpp [hunk 4]
```diff
@@ -47,11 +47,11 @@ class CfgPatches
 			"A3_Modules_F",
 			"A3_Modules_F_Curator",
             "A3_Modules_F_Orange",
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/modules_f_ares/config.cpp]: Weird spacing

File: @AresModAchillesExpansion/addons/functions_f/functions/common/fnc_printZeusMessage.sqf [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+	Description:
+		Displays the message in the top bar of the Zeus interface.
+	
+	Parameters:
+		_message	- <STRING> or <ARRAY> The message as a string or a format array.
+	
+	Returns:
+		none
+	
+	Exampes:
```
Comment (neilzar) [inline @AresModAchillesExpansion/addons/functions_f/functions/common/fnc_printZeusMessage.sqf]:
```suggestion
	Examples:
```

File: @AresModAchillesExpansion/addons/functions_f/functions/common/fnc_printZeusError.sqf [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	Description:
+		Displays the error message in the top bar of the Zeus interface.
+	
+	Parameters:
+		_message	- <STRING> or <ARRAY> The message as a string or a format array.
+	
+	Returns:
+		none
+	
+	Exampes:
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f/functions/common/fnc_printZeusError.sqf]:
```suggestion
	Examples:
```

File: @AresModAchillesExpansion/addons/functions_f/functions/module/fnc_init.sqf [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+		_hasSelectionOption	- <BOOL> [false] If true, then the selection option is used when the module is placed on nothing
+		_doMultiSelect		- <BOOL> [false] If true, then #selection will return an array of entities instead of a single entity
+	
+	Alternative Parameters (without selection):
+		_nextMode			- <string> The mode that is called next
+	
+	Returns:
+		none
+	
+	Exampes:
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f/functions/module/fnc_init.sqf]:
```suggestion
	Examples:
```

File: @AresModAchillesExpansion/addons/functions_f/functions/module/fnc_openDialog.sqf [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+			_default		- <SCALAR> or <STRING> or <ARRAY> [...] Default value
+			_forceDefault 	- <BOOL> [true] If true, the default value is enforced
+		_links 		- <ARRAY> The modes that will be called next
+			_onConfirmed	- <STRING> The mode that is called on pressing ok
+			_onCanceled		- <STRING> The mode that is called on pressing cancel
+	
+	Returns:
+		none
+	
+	Exampes:
```
Comment (CreepPork) [inline @AresModAchillesExpansion/addons/functions_f/functions/module/fnc_openDialog.sqf]:
```suggestion
	Examples:
```

**Pull request closed without merging**